src/main/java/revir/user/Parser.java

```diff
@@ -0,0 +1,144 @@
+package revir.user;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import revir.system.exceptions.IllegalCommandException;
+import revir.system.exceptions.InvalidFormatException;
+import revir.tasks.Deadline;
+import revir.tasks.Event;
+import revir.tasks.Todo;
+import revir.user.command.Command;
+import revir.user.command.Create;
+import revir.user.command.Delete;
+import revir.user.command.Find;
+import revir.user.command.ListTasks;
+import revir.user.command.Mark;
+import revir.user.command.Nop;
+
+/**
+ * Parses the command string and returns the corresponding Command object.
+ */
+public class Parser {
+    private enum Action {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, INVALID
+    }
+
+    private static DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    private static final Map<String, Action> actionMap = new HashMap<>();
```

Comment:
Should final variables be named in SCREAMING_SNAKE_CASE according to Java conventions?